<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Martin Mosbeck &lt;martin.mosbeck@gmx.at&gt;</author>
 <class>DialogSignalConfig</class>
 <widget class="QDialog" name="DialogSignalConfig">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>510</width>
    <height>258</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Signal Config </string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="5">
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="1">
    <widget class="QLabel" name="label_signal_name">
     <property name="text">
      <string>Signal name</string>
     </property>
     <property name="buddy">
      <cstring>signal_name</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="2" colspan="3">
    <widget class="QLineEdit" name="signal_name">
     <property name="autoFillBackground">
      <bool>true</bool>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QLabel" name="label_signal_type">
     <property name="text">
      <string>Signal type</string>
     </property>
     <property name="buddy">
      <cstring>signal_type</cstring>
     </property>
    </widget>
   </item>
   <item row="2" column="2" colspan="3">
    <widget class="QComboBox" name="signal_type">
     <property name="currentText">
      <string>std_logic_vector</string>
     </property>
     <item>
      <property name="text">
       <string>std_logic_vector</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>std_logic</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>signed</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>unsigned</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>custom</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="2" column="6">
    <widget class="QLineEdit" name="custom_signal_type">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="autoFillBackground">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <spacer name="horizontalSpacer_3">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="3" column="1">
    <widget class="QLabel" name="label_length_type">
     <property name="text">
      <string>Length type</string>
     </property>
     <property name="buddy">
      <cstring>option_fixed_length</cstring>
     </property>
    </widget>
   </item>
   <item row="3" column="2" colspan="2">
    <widget class="QRadioButton" name="option_fixed_length">
     <property name="text">
      <string>Fixed</string>
     </property>
    </widget>
   </item>
   <item row="3" column="4" colspan="2">
    <widget class="QRadioButton" name="option_variable_length">
     <property name="text">
      <string>Variable</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="6">
    <widget class="QRadioButton" name="option_single">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Single</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="7">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>9</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="4" column="1" colspan="2">
    <widget class="QLabel" name="label_length_placeholder">
     <property name="text">
      <string>Length Placeholder</string>
     </property>
     <property name="buddy">
      <cstring>length_placeholder</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="3" colspan="3">
    <widget class="QLineEdit" name="length_placeholder">
     <property name="autoFillBackground">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="5" column="1" colspan="2">
    <widget class="QLabel" name="label_resulting_definition">
     <property name="text">
      <string>Resulting definition</string>
     </property>
     <property name="buddy">
      <cstring>resulting_definition</cstring>
     </property>
    </widget>
   </item>
   <item row="5" column="3" colspan="4">
    <widget class="QLineEdit" name="resulting_definition">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="6" column="1" colspan="6">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="7" column="6">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="8" column="4">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogSignalConfig</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>553</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogSignalConfig</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>553</x>
     <y>312</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
